dotnet/src/Connectors/Connectors.Memory.MongoDB/MongoDBVectorStoreRecordCollection.cs

```diff
@@ -303,6 +303,63 @@ public virtual async Task<VectorSearchResults<TRecord>> VectorizedSearchAsync<TV
             cancellationToken).ConfigureAwait(false);
     }
 
+    /// <inheritdoc />
+    public async IAsyncEnumerable<TRecord> GetAsync(Expression<Func<TRecord, bool>> filter, int top,
+        FilterOptions<TRecord>? options = null, [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(filter);
+        Verify.NotLessThan(top, 1);
+
+        options ??= new();
+
+        // Translate the filter now, so if it fails, we throw immediately.
+        var translatedFilter = new MongoDBFilterTranslator().Translate(filter, this._model);
+        SortDefinition<BsonDocument>? sortDefinition = null;
+        if (options.Sort.Values.Count > 0)
+        {
+            sortDefinition = Builders<BsonDocument>.Sort.Combine(
+                options.Sort.Values.Select(pair =>
+                {
+                    var storageName = this._model.GetDataOrKeyProperty(pair.Key).StorageName;
+
+                    return pair.Value
```

Comment:
Would definitely be nice to have IsAscending here rather than Value (so ValueTuple or dedicated struct instead of KeyValuePair)